packages/webchannel-wrapper/gulpfile.js

```diff
@@ -0,0 +1,188 @@
+// Valid levels: WHITESPACE_ONLY, SIMPLE_OPTIMIZATIONS, ADVANCED_OPTIMIZATIONS.
+const OPTIMIZATION_LEVEL = 'ADVANCED_OPTIMIZATIONS';
+
+// For minified builds, wrap the output so we avoid leaking global variables.
+const CJS_WRAPPER_PREFIX = `(function() {var firebase = require('@firebase/app').default;`;
+const CJS_WRAPPER_SUFFIX =
+  `}).apply(typeof global !== 'undefined' ? ` +
+  `global : typeof self !== 'undefined' ? ` +
+  `self : typeof window !== 'undefined' ? window : {});`;
+
+const closureLibRoot = path.dirname(
+  require.resolve('google-closure-library/package.json')
+);
+
+const closureDefines = [
+  // Avoid unsafe eval() calls (https://github.com/firebase/firebase-js-sdk/issues/798)
+  'goog.json.USE_NATIVE_JSON=true',
+  // Disable debug logging (saves 8780 bytes).
+  'goog.DEBUG=false',
+  // Disable fallbacks for running async code (saves 1472 bytes).
+  'goog.ASSUME_NATIVE_PROMISE=true',
+  // Disables IE8-specific event fallback code (saves 523 bytes).
+  'goog.events.CAPTURE_SIMULATION_MODE=0',
+  // Disable IE-Specific ActiveX fallback for XHRs (saves 524 bytes).
+  'goog.net.XmlHttpDefines.ASSUME_NATIVE_XHR=true'
+];
+
+/**
+ * Generates a closure compiler build of webchannel-wrapper.
+ * @param {string} filename name of the generated file
+ * @param {string} prefix prefix to the compiled code
+ * @param {string} suffix suffix to the compiled code
+ */
+function createBuildTask(filename, prefix, suffix) {
+  return function closureBuild() {
+    return gulp
+      .src(
+        [
+          `${closureLibRoot}/closure/goog/**/*.js`,
+          `${closureLibRoot}/third_party/closure/goog/**/*.js`,
+          'src/**/*.js'
+        ],
+        { base: '.' }
+      )
+      .pipe(sourcemaps.init())
+      .pipe(
+        closureCompiler({
+          js_output_file: filename,
+          output_wrapper: `${prefix}%output%${suffix}`,
+          entry_point: 'firebase.webchannel.wrapper',
+          compilation_level: OPTIMIZATION_LEVEL,
+          externs: [
+            resolve(__dirname, './externs/overrides.js'),
+            resolve(__dirname, './externs/module.js')
+          ],
+          language_out: 'ECMASCRIPT_2017',
+          dependency_mode: 'PRUNE',
+          define: closureDefines
+        })
+      )
+      .pipe(sourcemaps.write('.'))
+      .pipe(gulp.dest('dist'));
+  };
+}
+
+function createRollupTask({
+  inputPath,
+  outputExtension,
+  compileToES5 = false,
+  format = 'es'
+}) {
+  return async function rollupBuild() {
+    const plugins = [rollupSourcemaps(), commonjs()];
+    if (compileToES5) {
+      plugins.push(
+        typescriptPlugin({
+          typescript,
+          // Uncomment for development only. Prevents caching between builds.
+          // clean: true,
+          compilerOptions: { allowJs: true },
+          include: ['dist/**/*.js']
+        })
+      );
+    }
+    // Typescript step unminifies minified Closure output.
+    plugins.push(terser());
```

Comment:
It looks bad in terms of package size increase and technically increases the size of the npm download.

Reply:
Is it just removing whitespaces, I guess Typescript just reformats the already mangled code?
Anyway I don't think keeping those numbers low are that meaningful to the SDK, and it will slow down the build.

Reply:
Removed Terser.